source/grid/grid_tools.cc

```diff
@@ -6452,6 +6458,479 @@ namespace GridTools
 
       return result;
     }
+
+
+
+    template <int structdim, int spacedim>
+    template <int dim>
+    DistributedComputePointLocationsInternal<dim, spacedim>
+    DistributedComputeIntersectionLocationsInternal<structdim, spacedim>::
+      convert_to_distributed_compute_point_locations_internal(
+        const unsigned int                  n_quadrature_pnts,
+        const Triangulation<dim, spacedim> &tria,
+        const Mapping<dim, spacedim> &      mapping,
+        const bool consistent_numbering_of_sender_and_receiver) const
+    {
+      using CellIterator =
+        typename Triangulation<dim, spacedim>::active_cell_iterator;
+
+      GridTools::internal::DistributedComputePointLocationsInternal<dim,
+                                                                    spacedim>
+        result;
+
+      // We need quadrature rules for the intersections. We are using a
+      // QGaussSimplex quadrature rule since CGAL always returns simplices
+      // as intersections.
+      const QGaussSimplex<structdim> quadrature(n_quadrature_pnts);
+
+      // Resulting quadrature points get different indices. In the case the
+      // requested intersections are unique also the resulting quadrature
+      // points are unique and we can simply number the points in an
+      // ascending way.
+      for (auto const &recv_component : recv_components)
+        {
+          // dependent on the size of the intersection an empty quadrature
+          // is returned. Therefore, we have to compute the quadrature also
+          // here.
+          const Quadrature<spacedim> &quad =
+            quadrature.compute_affine_transformation(
+              std::get<2>(recv_component));
+
+          for (unsigned int i = 0; i < quad.size(); ++i)
+            {
+              // the third component of result.recv_components is not needed
+              // before finalize_setup.
+              result.recv_components.emplace_back(
+                std::get<0>(recv_component),
+                result.recv_components.size(), // number of point
+                numbers::invalid_unsigned_int);
+            }
+        }
+
+      // since empty quadratures might be present we have to set the number
+      // of searched points after inserting the point indices into
+      // recv_components
+      result.n_searched_points = result.recv_components.size();
+
+#ifdef DEAL_II_WITH_MPI
```

Comment:
Do we need these ifdefs? The checks could be done in `communicate_indices`.

Reply:
[f752268](https://github.com/dealii/dealii/pull/15165/commits/f7522682d88706664d6d68c5347af6544d64897d)